ql/src/java/org/apache/hadoop/hive/ql/ddl/database/create/CreateDatabaseDesc.java

```diff
@@ -45,13 +46,14 @@ public class CreateDatabaseDesc implements DDLDesc, Serializable {
   private final String remoteDbName;
   private final Map<String, String> dbProperties;
 
-  public CreateDatabaseDesc(String databaseName, String comment, String locationUri, String managedLocationUri,
+  public CreateDatabaseDesc(String catalogName, String databaseName, String comment, String locationUri, String managedLocationUri,
       boolean ifNotExists, Map<String, String> dbProperties) {
-    this(databaseName, comment, locationUri, managedLocationUri, ifNotExists, dbProperties, "NATIVE", null, null);
+    this(catalogName, databaseName, comment, locationUri, managedLocationUri, ifNotExists, dbProperties, "NATIVE", null, null);
```

Comment:
for the external catalogs `dbtype` should be "REMOTE", right? It seems that we always hardcode "NATIVE" and below condition never returns true
````
 if (dbtype != null && dbtype.equalsIgnoreCase("REMOTE")) {
````

Reply:
> should we introduce a flag like native/non-native similar to tables?

This can be done when we implement specific catalogs (JDBC catalog & Iceberg catalog, etc.).